routers/api.py

```
@@ -133,175 +130,185 @@ def script_to_api(page_cls: typing.Type[BasePage]):
-    ):
-        # parse form data
-        page_request = _parse_form_data(
-            request_model,
-            form_data,
-            page_request_json,
-            workspace=api_key.workspace,
-            user=api_key.created_by,
+        def run_api_json(
+            request: Request,
+            page_request: request_model,
+            api_key: ApiKey = Depends(api_auth_header),
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)